test/boost/secondary_index_test.cc

```diff
@@ -1208,6 +1208,279 @@ SEASTAR_TEST_CASE(test_indexing_paging_and_aggregation) {
     });
 }
 
+// Verifies that both "SELECT * [rest_of_query]" and "SELECT count(*) [rest_of_query]" 
+// return expected count of rows.
+void assert_select_count_and_select_rows_has_size(cql_test_env& e, const sstring& rest_of_query, int64_t expected_count) {
+    eventually([&] { 
+        auto res = e.execute_cql("SELECT count(*) " + rest_of_query).get0();
+        assert_that(res).is_rows().with_rows({
+            { long_type->decompose(expected_count) }
+        });
+        res = e.execute_cql("SELECT * " + rest_of_query).get0();
```

Comment:
`cquery_nofail` would be better here, for improved diagnostic in case of failure.